### PR TITLE
CloudWatch: Add missing AWS/Prometheus metrics

### PR DESCRIPTION
**Pull request by jangaraj**
Reference: https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-CW-usage-metrics.html

### CODE REVIEW

File: pkg/tsdb/cloudwatch/metrics.go [hunk 1]
```diff
@@ -368,6 +368,7 @@ var metricsMap = map[string][]string{
 	"AWS/Polly":                {"2XXCount", "4XXCount", "5XXCount", "RequestCharacters", "ResponseLatency"},
 	"AWS/PrivateLinkEndpoints": {"ActiveConnections", "BytesProcessed", "NewConnections", "PacketsDropped", "RstPacketsReceived"},
 	"AWS/PrivateLinkServices":  {"ActiveConnections", "BytesProcessed", "EndpointsCount", "NewConnections", "RstPacketsReceived"},
+	"AWS/Prometheus":           {"AlertManagerAlertsReceived", "AlertManagerNotificationsFailed", "AlertManagerNotificationsThrottled", "DiscardedSamples", "RuleEvaluations", "RuleEvaluationFailures", "RuleGroupIterationsMissed"},
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/metrics.go:371]: Where did `AlertManagerNotificationsFailed` come from? I didn't see it in the linked docs.

File: pkg/tsdb/cloudwatch/metrics.go [hunk 2]
```diff
@@ -484,6 +485,7 @@ var dimensionsMap = map[string][]string{
 	"AWS/Polly":                   {"Operation"},
 	"AWS/PrivateLinkEndpoints":    {"Endpoint Type", "Service Name", "Subnet Id", "VPC Endpoint Id", "VPC Id"},
 	"AWS/PrivateLinkServices":     {"Az", "Load Balancer Arn", "Service Id", "VPC Endpoint Id"},
+	"AWS/Prometheus":              {"Reason", "RuleGroup", "Workspace"},
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/metrics.go:488]: Also did not see these in the linked docs.

**Review by iwysiu — COMMENTED** [review]
Hi @jangaraj ! Thanks for the pr. I have questions about where a couple of the metrics came from, but otherwise it looks good!

File: pkg/tsdb/cloudwatch/metrics.go (hunk 1 above)
Comment (jangaraj) [reply pkg/tsdb/cloudwatch/metrics.go:371]: Good catch and expected question. AWS doc is not always good and this is an example. `AlertManagerNotificationsFailed` is valid metrics (I requested also AWS to update their doc):
image

**Review by iwysiu — APPROVED** [review]
Thanks for the screenshots! Nice work.

**Review by sarahzinger — APPROVED** [review]
the ci here is failing, but I believe this is due to an issue where external contributors do not have access to the tokens necessary to run our ci checks, and not due to the code itself. 

Thanks for the contribution! 🎉

**Pull request merged**